spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/AbstractMessageListenerContainer.java

```diff
@@ -637,11 +623,10 @@ public void setForceCloseChannel(boolean forceCloseChannel) {
 	 */
 	protected String getRoutingLookupKey() {
 		return super.getConnectionFactory() instanceof RoutingConnectionFactory
-				? (this.lookupKeyQualifier + this.queues.stream()
+				? (this.lookupKeyQualifier + "[" + this.queues.stream()
 								.map(q -> q.getName())
-								.collect(Collectors.toList())
-							.toString()
-							.replaceAll(" ", ""))
+								.collect(Collectors.joining(",")) + "]")
+						.replaceAll(" ", "")
```

Comment:
Why do we need to replace whitespaces if there is no one anymore? You join over just `,`. No `toString()` leftovers also.

Reply:
Because that's how we tell them to configure the keys on the factory...

>Also starting with version 1.4, you can configure a routing connection factory in a listener container. In that case, the list of queue names is used as the lookup key. For example, if you configure the container with setQueueNames("foo", "bar"), the lookup key will be "[foo,bar]" (no spaces).

Reply:
Right, but I still don't understand from where those spaces come since we have changed algorithm from `List.toString()` to the `Collectors.joining(",")`.

:man_shrugging:

Reply:
Hehe - there was a bug in the test

    container.setQueueNames("foo, bar");

instead of

    container.setQueueNames("foo", "bar");

So that's where the space came from 😄